src/version.cpp

```diff
@@ -252,117 +249,133 @@ void Exiv2::dumpLibraryInfo(std::ostream& os,const exv_grep_keys_t& keys)
     "unknown";
 #endif
 
-    int have_gmtime_r    =0;
-    int have_libintl     =0;
-    int have_lensdata    =0;
-    int have_iconv       =0;
-    int have_lstat       =0;
-    int have_stdlib      =0;
-    int have_strlib      =0;
-    int have_strerror_r  =0;
-    int have_strings_h   =0;
-    int have_mmap        =0;
-    int have_munmap      =0;
-    int have_unistd_h    =0;
-    int have_sys_mman    =0;
-    int have_libz        =0;
-    int have_xmptoolkit  =0;
-    int adobe_xmpsdk     =0;
-    int have_bool        =0;
-    int have_unistd      =0;
-    int have_unicode_path=0;
-    int have_regex       =0;
-
-    int enable_webready  =0;
-    int enable_nls       =0;
-    int use_curl         =0;
+    constexpr int have_strlib = 0;
+    constexpr int have_strings_h = 0;
 
 #ifdef EXV_HAVE_GMTIME_R
-    have_gmtime_r=1;
+    constexpr int have_gmtime_r = 1;
+#else
+    constexpr int have_gmtime_r = 0;
 #endif
 
 #ifdef EXV_HAVE_LIBINTL_H
-    have_libintl=1;
+    constexpr int have_libintl = 1;
+#else
+    constexpr int have_libintl = 0;
 #endif
 
 #ifdef EXV_HAVE_LENSDATA
-    have_lensdata=1;
+    constexpr int have_lensdata = 1;
+#else
+    constexpr int have_lensdata = 0;
 #endif
 
 #ifdef EXV_HAVE_ICONV
-    have_iconv=1;
-#endif
-
-#ifdef EXV_HAVE_LIBINTL_H
-    have_libintl=1;
+    constexpr int have_iconv = 1;
+#else
+    constexpr int have_iconv = 0;
 #endif
 
 #ifdef EXV_HAVE_LSTAT
-    have_lstat=1;
+    constexpr int have_lstat = 1;
+#else
+    constexpr int have_lstat = 0;
 #endif
 
 #ifdef EXV_HAVE_STDLIB_H
-    have_stdlib=1;
+    constexpr int have_stdlib = 1;
```

Comment:
Are we benefiting anyhow by using in these variables `constexpr` instead of just `const` ?

Reply:
Yes, it is like a #define now and evaluated at compile time.